src/apps/tools.clj

```diff
@@ -96,19 +108,24 @@
   [{:keys [tools]}]
   (transaction
     (let [tool-ids (doall (map add-new-tool tools))]
+      (dorun (map permissions/register-public-tool tool-ids))
```

Comment:
Yeah, I ended up doing something similar for apps. It probably is better to err on the side of not saving things in the apps database if the permissions can't be created.